// @cnv.tech/cnv-design/dist/Components/Inputs/Radio/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const utils_1 = require("../../../utils");
require("./index.css");
const _name = 'radio';
const Radio = props => {
    const inputRef = react_1.default.useRef(null);
    let { children } = props;
    let attrs = getAttrs(props);
    let attrsInput = getAttrsInput(props, inputRef);
    return react_1.default.createElement("label", Object.assign({}, attrs),
        react_1.default.createElement("input", Object.assign({ ref: inputRef }, attrsInput)),
        children);
};
const getAttrs = (props) => {
    let { disabled, className } = props;
    return {
        className: utils_1.utils.getClassNames(utils_1.utils.getName(_name), {
            'disabled': disabled
        }, className)
    };
};
const getAttrsInput = (props, ref) => {
    let { name, value, checked, disabled } = props;
    return {
        type: 'radio',
        name: name,
        value: value,
        checked: checked !== null && checked !== void 0 ? checked : false,
        disabled: disabled,
        onClick: e => onClickHandler(e, props, ref)
    };
};
const onClickHandler = (e, props, ref) => {
    let { onChange } = props;
    if (onChange && ref.current)
        onChange(e, ref.current.value);
};
exports.default = Radio;
